# A buffer overflow vulnerability exists in multiple implementations of DNS
# resolver libraries.  This affects glibc-2.2.5 and earlier.

http://bugs.gentoo.org/show_bug.cgi?id=4923
http://www.cert.org/advisories/CA-2002-19.html
--- libc/resolv/nss_dns/dns-network.c	Fri Jul 12 10:18:13 2002
+++ libc/resolv/nss_dns/dns-network.c	Fri Jul 12 10:20:10 2002
@@ -328,7 +328,9 @@
 	    }
 	  cp += n;
 	  *alias_pointer++ = bp;
-	  bp += strlen (bp) + 1;
+	  n = strlen (bp) + 1;
+	  bp += n;
+	  linebuflen -= n;
 	  result->n_addrtype = class == C_IN ? AF_INET : AF_UNSPEC;
 	  ++have_answer;
 	}
